nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/blob/ColdStorageHelper.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * (C) Copyright 2020 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Salem Aouana
+ */
+
+package org.nuxeo.ecm.core.blob;
+
+import static org.nuxeo.ecm.core.schema.FacetNames.COLD_STORAGE;
+
+import java.io.Serializable;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.nuxeo.ecm.core.api.Blob;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.DocumentRef;
+import org.nuxeo.ecm.core.api.thumbnail.ThumbnailService;
+import org.nuxeo.runtime.api.Framework;
+
+/**
+ * Manages the cold storage of the main content of a {@link DocumentModel}.
+ *
+ * @since 11.1
+ */
+public class ColdStorageHelper {
+
+    private static final Logger log = LogManager.getLogger(ColdStorageHelper.class);
+
+    public static final String FILE_CONTENT_PROPERTY = "file:content";
+
+    public static final String COLD_STORAGE_CONTENT_PROPERTY = "coldstorage:coldContent";
+
+    /**
+     * Moves the main content associated with the document of the given {@link DocumentRef} to a cold storage.
+     */
+    public static boolean moveContentToColdStorage(CoreSession session, DocumentRef documentRef) {
+        DocumentModel document = session.getDocument(documentRef);
+        if (document.hasFacet(COLD_STORAGE)) {
+            log.debug("The main content of document: {} is already in cold storage", document);
+            return false;
+        }
```

Comment:
I think the logic is not correct.
We should have:
```
if (document.hasFacet(COLD_STORAGE) && document.getPropertyValue(COLD_STORAGE_CONTENT_PROPERTY) != null) {
    log.debug("The main content of document: {} is already in cold storage", document);
    return false;

Reply:
yes,  i was just waiting for https://github.com/nuxeo/nuxeo/pull/3666#discussion_r368565715 before making the changes it's why i didn't include the second part of the check

Reply:
i have my answer i will include it on my next push after your final review ;)